security/acme-client/src/opnsense/mvc/app/views/OPNsense/AcmeClient/actions.volt

```diff
@@ -60,7 +60,135 @@ POSSIBILITY OF SUCH DAMAGE.
+
+                        if (status === "success" && data.status === "ok")
+                            identityDiv.text(data.identity).show();
+                        else
+                            identityDiv.text("{{ lang._('Failed loading identity') }}").show();
+                    });
+                });
+
+            // Hide when input changes that influences the identity.
+            $identityType.change(function() {
+                identityDiv.hide();
+            });
+        })($('#action\\.sftp_identity_type'));
+
+        // SFTP - Connection test button
+        (function ($user) {
+            var statusDiv = makeStatusDiv($user, 'alert-success').html(
+                '<div class="message"></div>'
+                + '<div class="detail-enabler" style="cursor: pointer"><i class="fa fa-plus-square"></i></div>'
+                + '<div class="detail" style="font-family: monospace"></div>');
+
+            statusDiv.find(".detail-enabler").click(function() {
+                $(".detail", statusDiv).show();
+                $(this).hide();
+            });
+
+            var errors = [
+                {cond: ["connect_failed", "invalid_parameters"], msg: "{{ lang._('Host or username not specified.') }}"},
+                {cond: ["connect_failed", "host_not_trusted"], msg: "{{ lang._('Host cannot be trusted.') }}"},
+                {cond: ["connect_failed", "host_not_resolved"], msg: "{{ lang._('Failed to resolve hostname.') }}"},
+                {cond: ["connect_failed", "connection_refused"], msg: "{{ lang._('Connection to host refused.') }}"},
+                {cond: ["connect_failed", "network_timeout"], msg: "{{ lang._('Connection timed out.') }}"},
+                {cond: ["connect_failed", "network_unreachable"], msg: "{{ lang._('Host not reachable.') }}"},
+                {cond: ["connect_failed", "permission_denied"], msg: "{{ lang._('Host does not permit a connection for the specified user & identity.') }}"},
+                {cond: ["connect_failed"], msg: "{{ lang._('Failed to connect to host.') }}"},
+                {cond: ["change_home_dir_failed"], msg: "{{ lang._('Failed to change the remote path.') }}"},
+                {cond: ["permission_denied"], msg: "{{ lang._('Uploads are not allowed to the specified remote path.') }}"},
+                {msg: "{{ lang._('Test failed, see details.') }}"},
+            ];
+
+            makeButton("{{ lang._('Test Connection') }}", "upload_sftp")
+                .click(function () {
+                    statusDiv.hide();
+                    var button = $(this);
+                    button.prop('disabled', true).find(".fa-spinner").show();
+
+                    ajaxCall("/api/acmeclient/actions/sftpTestConnection", getFormData("DialogAction").action, function (data, status) {
+                        button.prop('disabled', false).find(".fa-spinner").hide();
+
+                        var message = "", detail = "", statusClass = "alert-warning";
```

Comment:
I would write each of them in its own line.

Reply:
ok